### PR TITLE
[Flax] Adapt flax examples to include `push_to_hub`

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR adapts all Flax examples to automatically push trained checkpoints to the hub

### CODE REVIEW

File: examples/flax/language-modeling/README.md [hunk 1]
```diff
@@ -33,11 +33,37 @@ in Norwegian on a single TPUv3-8 pod.
...
+
+```
+git clone https://huggingface.co/<your-username>/norwegian-roberta-base
+```
+
+To ensure that all tensorboard traces will be uploaded correctly, we need to 
+track them. You can run the following command inside your model repo to do so.
+
+```
+cd norwegian-roberta-base
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:56]: This is a bit hacky, but it's there is no `git_lfs_track` abstraction in `transformers` yet. Think for the sprint, it's ok to leave the somewhat hacky solution for now.

**Comment by patrickvonplaten** [comment]
@patil-suraj, I think we should also add a `README.md` for the summarization example (this will probs be used a lot during the sprint).

File: examples/flax/text-classification/run_flax_glue.py [hunk 2]
```diff
@@ -123,6 +123,11 @@ def parse_args():
     )
     parser.add_argument("--output_dir", type=str, default=None, help="Where to store the final model.")
     parser.add_argument("--seed", type=int, default=3, help="A seed for reproducible training.")
+    parser.add_argument(
+        "--disable_push_to_hub",
```
Comment (patil-suraj) [inline examples/flax/text-classification/run_flax_glue.py:127]: why not use `push_to_hub` here?

**Review by patil-suraj — APPROVED** [review]
LGTM!

File: examples/flax/language-modeling/run_clm_flax.py [hunk 3]
```diff
@@ -604,10 +604,10 @@ def eval_step(params, batch):
...
             write_metric(summary_writer, train_metrics, eval_metrics, train_time, cur_step)
 
-    # save last checkpoint
-    if jax.process_index() == 0:
-        params = jax.device_get(unreplicate(state.params))
-        model.save_pretrained(training_args.output_dir, params=params)
+        # save checkpoint after each epoch and push checkpoint to the hub
+        if jax.process_index() == 0:
+            params = jax.device_get(unreplicate(state.params))
+            model.save_pretrained(training_args.output_dir, params=params, push_to_hub=training_args.push_to_hub)
```
Comment (sgugger) [inline examples/flax/language-modeling/run_clm_flax.py:610]: You should add a commit message here with the epoch number. (The arg name is `commit_message`).

File: examples/flax/language-modeling/run_mlm_flax.py [hunk 4]
```diff
@@ -642,9 +642,9 @@ def eval_step(params, batch):
...
+            write_metric(summary_writer, train_metrics, eval_metrics, train_time, cur_step)
 
-    # save last checkpoint
-    if jax.process_index() == 0:
-        params = jax.device_get(jax.tree_map(lambda x: x[0], state.params))
-        model.save_pretrained(training_args.output_dir, params=params)
+        # save checkpoint after each epoch and push checkpoint to the hub
+        if jax.process_index() == 0:
+            params = jax.device_get(jax.tree_map(lambda x: x[0], state.params))
+            model.save_pretrained(training_args.output_dir, params=params, push_to_hub=training_args.push_to_hub)
```
Comment (sgugger) [inline examples/flax/language-modeling/run_mlm_flax.py:650]: Same here.

File: examples/flax/summarization/run_summarization_flax.py [hunk 5]
```diff
@@ -787,10 +787,10 @@ def generate_step(params, batch):
...
         logger.info(desc)
 
-    # save last checkpoint
-    if jax.process_index() == 0:
-        params = jax.device_get(unreplicate(state.params))
-        model.save_pretrained(training_args.output_dir, params=params)
+        # save checkpoint after each epoch and push checkpoint to the hub
+        if jax.process_index() == 0:
+            params = jax.device_get(jax.tree_map(lambda x: x[0], state.params))
+            model.save_pretrained(training_args.output_dir, params=params, push_to_hub=training_args.push_to_hub)
```
Comment (sgugger) [inline examples/flax/summarization/run_summarization_flax.py:793]: And here

File: examples/flax/text-classification/run_flax_glue.py [hunk 6]
```diff
@@ -491,10 +496,10 @@ def eval_step(state, batch):
...
         write_metric(train_metrics, eval_metric, train_time, cur_step)
 
-    # save last checkpoint
-    if jax.process_index() == 0:
-        params = jax.device_get(jax.tree_map(lambda x: x[0], state.params))
-        model.save_pretrained(args.output_dir, params=params)
+        # save checkpoint after each epoch and push checkpoint to the hub
+        if jax.process_index() == 0:
+            params = jax.device_get(jax.tree_map(lambda x: x[0], state.params))
+            model.save_pretrained(args.output_dir, params=params, push_to_hub=args.push_to_hub)
```
Comment (sgugger) [inline examples/flax/text-classification/run_flax_glue.py:502]: And there!

**Review by sgugger — APPROVED** [review]
Amazing to add this to all models! There is a Python solution for the TensorBoard logs (see offline) that you can use, otherwise it looks great!

File: examples/flax/language-modeling/README.md [hunk 7]
```diff
@@ -33,11 +33,37 @@ in Norwegian on a single TPUv3-8 pod.
...
+echo "*.v2 filter=lfs diff=lfs merge=lfs -text" >> .gitattributes
```
Comment (julien-c) [inline examples/flax/language-modeling/README.md:57]:
```suggestion
git lfs track "*tfevents*"
```

File: examples/flax/language-modeling/README.md (hunk 1 above)
Comment (LysandreJik) [reply examples/flax/language-modeling/README.md:56]: Alternatively you could do it like that with huggingface-hub v0.0.13:

```py
from huggingface_hub.repository import Repository
from transformers import BertModel, BertConfig

model = BertModel(BertConfig(num_hidden_layers=1, num_attention_heads=2, hidden_size=2, intermediate_size=2, vocab_size=10))

repo = Repository("repo-id", clone_from="<user>/repo-id", use_auth_token=True)

with repo.commit("Track tfevents"):
    repo.lfs_track("*tfevents*")

with repo.commit("New model version"):
    model.save_pretrained(".")
```

**Review by LysandreJik — APPROVED** [review]

File: examples/flax/language-modeling/README.md (hunk 1 above)
Comment (patrickvonplaten) [reply examples/flax/language-modeling/README.md:56]: going for the simple `git lfs track "*tfevents*"` solution for new. 

The user needs to pass the `"user-name"` which would require some changes to the scripts. Think we can do this in a follow-up PR.
